server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2521,6 +2521,13 @@ public UserVm updateVirtualMachine(UpdateVMCmd cmd) throws ResourceUnavailableEx
         String extraConfig = cmd.getExtraConfig();
 
         UserVmVO vmInstance = _vmDao.findById(cmd.getId());
+        if (MapUtils.isNotEmpty(details)) {
```

Comment:
@Spaceman1984 should we also check cleanupDetails param here?

Reply:
Yes, that would be good.